oca/api.py

```diff
@@ -53,45 +53,31 @@ def method(self):
             instantiated with an iterable of the decorated method results.
 
     Raises:
-        ValueError: If no container type is provided, and a value is returned
-            from the decorated method.
-        TypeError: If an incorrect data type is provided as the `container`.
-            Valid arguments are `None`, instances of `type`, or a callable.
+        AssertionError: If no container type is provided, and a value is
+            returned from the decorated method. This will only be raised if
+            ``PYTHONOPTIMIZE`` is less than 2.
+        AssertionError: If an incorrect data type is provided as `container`.
+            Valid arguments are `None` or instances of `type`. This will only
```

Comment:
(Following from suggestions below:) This would be then a `TypeError`, and any callable would be a valid value.